catroid/src/main/java/org/catrobat/catroid/content/BroadcastHandler.java

```diff
@@ -51,48 +51,52 @@ private BroadcastHandler() {
 		throw new AssertionError();
 	}
 
-	public static void doHandleBroadcastEvent(Look look, String broadcastMessage) {
+	public static void doHandleBroadcastEvent(Look look, BroadcastEvent event, String broadcastMessage) {
 		String sceneName = ProjectManager.getInstance().getSceneToPlay().getName();
-		if (!BroadcastSequenceMap.containsKey(broadcastMessage, sceneName)) {
+		BroadcastSequenceMap broadcastSequenceMap = look.sprite.getBroadcastSequenceMap();
+		if (broadcastSequenceMap.get(broadcastMessage, sceneName).isEmpty()) {
 			return;
 		}
 
-		for (SequenceAction action : BroadcastSequenceMap.get(broadcastMessage, sceneName)) {
+		for (SequenceAction action : broadcastSequenceMap.get(broadcastMessage, sceneName)) {
 			Script scriptOfAction = actionScriptMap.get(action);
 
 			if (!handleAction(action, scriptOfAction)) {
 				addOrRestartAction(look, action);
 			}
 		}
 
-		if (BroadcastWaitSequenceMap.containsKey(broadcastMessage, sceneName)) {
-			for (SequenceAction action : BroadcastWaitSequenceMap.get(broadcastMessage, sceneName)) {
+		if (event.getSenderSprite().getBroadcastWaitSequenceMap().containsKey(broadcastMessage, sceneName, look.sprite)) {
```

Comment:
this long dig: event.getSenderSprite().getBroadcastWaitSequenceMap(). show up a lot.
maybe make a local variable for it and rename the broadcastSequenceMap and this new one so that its clear whos who.